### DISCUSSION TITLE
Error while installing detectron2

### DISCUSSION THREAD

**Discussion by RANA-ATI** [facebookresearch/detectron2#4729] [Q&A]
I need to install detectron2 on my system. I usually run at collab but I recently bought a GPU so it's my first time installing over a local system. Below is the issue I am facing and I am not sure whats the cause.

(detectron) D:\University\FYP\OUR FYP\Model Training\Non Augmented Dataset MODELS\Model3 Detectron2 412>python -m pip install -e detectron2
Obtaining file:///D:/University/FYP/OUR%20FYP/Model%20Training/Non%20Augmented%20Dataset%20MODELS/Model3%20Detectron2%20412/detectron2
  Preparing metadata (setup.py) ... done
Requirement already satisfied: Pillow>=7.1 in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (9.3.0)
Requirement already satisfied: matplotlib in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (3.6.2)
Requirement already satisfied: pycocotools>=2.0.2 in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (2.0.6)
Requirement already satisfied: termcolor>=1.1 in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (2.1.1)
Requirement already satisfied: yacs>=0.1.8 in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (0.1.8)
Requirement already satisfied: tabulate in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (0.9.0)
Requirement already satisfied: cloudpickle in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (2.2.0)
Requirement already satisfied: tqdm>4.29.0 in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (4.64.1)
Requirement already satisfied: tensorboard in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (2.11.0)
Requirement already satisfied: fvcore<0.1.6,>=0.1.5 in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (0.1.5.post20221221)
Requirement already satisfied: iopath<0.1.10,>=0.1.7 in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (0.1.9)
Requirement already satisfied: omegaconf>=2.1 in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (2.3.0)
Requirement already satisfied: hydra-core>=1.1 in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (1.3.1)
Requirement already satisfied: black in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (22.12.0)
Requirement already satisfied: timm in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (0.6.12)
Requirement already satisfied: packaging in d:\anaconda\envs\detectron\lib\site-packages (from detectron2==0.6) (22.0)
Requirement already satisfied: pyyaml>=5.1 in d:\anaconda\envs\detectron\lib\site-packages (from fvcore<0.1.6,>=0.1.5->detectron2==0.6) (5.1)
Requirement already satisfied: numpy in d:\anaconda\envs\detectron\lib\site-packages (from fvcore<0.1.6,>=0.1.5->detectron2==0.6) (1.24.1)
Requirement already satisfied: antlr4-python3-runtime==4.9.* in d:\anaconda\envs\detectron\lib\site-packages (from hydra-core>=1.1->detectron2==0.6) (4.9.3)
Requirement already satisfied: importlib-resources in d:\anaconda\envs\detectron\lib\site-packages (from hydra-core>=1.1->detectron2==0.6) (5.10.2)
Requirement already satisfied: portalocker in d:\anaconda\envs\detectron\lib\site-packages (from iopath<0.1.10,>=0.1.7->detectron2==0.6) (2.6.0)
Requirement already satisfied: pyparsing>=2.2.1 in d:\anaconda\envs\detectron\lib\site-packages (from matplotlib->detectron2==0.6) (3.0.9)
Requirement already satisfied: kiwisolver>=1.0.1 in d:\anaconda\envs\detectron\lib\site-packages (from matplotlib->detectron2==0.6) (1.4.4)
Requirement already satisfied: cycler>=0.10 in d:\anaconda\envs\detectron\lib\site-packages (from matplotlib->detectron2==0.6) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in d:\anaconda\envs\detectron\lib\site-packages (from matplotlib->detectron2==0.6) (4.38.0)
Requirement already satisfied: python-dateutil>=2.7 in d:\anaconda\envs\detectron\lib\site-packages (from matplotlib->detectron2==0.6) (2.8.2)
Requirement already satisfied: contourpy>=1.0.1 in d:\anaconda\envs\detectron\lib\site-packages (from matplotlib->detectron2==0.6) (1.0.6)
Requirement already satisfied: colorama in d:\anaconda\envs\detectron\lib\site-packages (from tqdm>4.29.0->detectron2==0.6) (0.4.6)
Requirement already satisfied: platformdirs>=2 in d:\anaconda\envs\detectron\lib\site-packages (from black->detectron2==0.6) (2.5.2)
Requirement already satisfied: typing-extensions>=3.10.0.0 in d:\anaconda\envs\detectron\lib\site-packages (from black->detectron2==0.6) (4.4.0)
Requirement already satisfied: pathspec>=0.9.0 in d:\anaconda\envs\detectron\lib\site-packages (from black->detectron2==0.6) (0.10.3)
Requirement already satisfied: tomli>=1.1.0 in d:\anaconda\envs\detectron\lib\site-packages (from black->detectron2==0.6) (2.0.1)
Requirement already satisfied: click>=8.0.0 in d:\anaconda\envs\detectron\lib\site-packages (from black->detectron2==0.6) (8.1.3)
Requirement already satisfied: mypy-extensions>=0.4.3 in d:\anaconda\envs\detectron\lib\site-packages (from black->detectron2==0.6) (0.4.3)
Requirement already satisfied: setuptools>=41.0.0 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (65.5.0)
Requirement already satisfied: google-auth<3,>=1.6.3 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (2.15.0)
Requirement already satisfied: markdown>=2.6.8 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (3.4.1)
Requirement already satisfied: requests<3,>=2.21.0 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (2.28.1)
Requirement already satisfied: tensorboard-data-server<0.7.0,>=0.6.0 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (0.6.1)
Requirement already satisfied: werkzeug>=1.0.1 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (2.2.2)
Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (1.8.1)
Requirement already satisfied: absl-py>=0.4 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (1.3.0)
Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (0.4.6)
Requirement already satisfied: protobuf<4,>=3.9.2 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (3.20.3)
Requirement already satisfied: wheel>=0.26 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (0.37.1)
Requirement already satisfied: grpcio>=1.24.3 in d:\anaconda\envs\detectron\lib\site-packages (from tensorboard->detectron2==0.6) (1.51.1)
Requirement already satisfied: torch>=1.7 in d:\anaconda\envs\detectron\lib\site-packages (from timm->detectron2==0.6) (1.8.0+cu101)
Requirement already satisfied: huggingface-hub in d:\anaconda\envs\detectron\lib\site-packages (from timm->detectron2==0.6) (0.11.1)
Requirement already satisfied: torchvision in d:\anaconda\envs\detectron\lib\site-packages (from timm->detectron2==0.6) (0.9.0+cu101)
Requirement already satisfied: pyasn1-modules>=0.2.1 in d:\anaconda\envs\detectron\lib\site-packages (from google-auth<3,>=1.6.3->tensorboard->detectron2==0.6) (0.2.8)
Requirement already satisfied: six>=1.9.0 in d:\anaconda\envs\detectron\lib\site-packages (from google-auth<3,>=1.6.3->tensorboard->detectron2==0.6) (1.16.0)
Requirement already satisfied: rsa<5,>=3.1.4 in d:\anaconda\envs\detectron\lib\site-packages (from google-auth<3,>=1.6.3->tensorboard->detectron2==0.6) (4.9)
Requirement already satisfied: cachetools<6.0,>=2.0.0 in d:\anaconda\envs\detectron\lib\site-packages (from google-auth<3,>=1.6.3->tensorboard->detectron2==0.6) (5.2.0)
Requirement already satisfied: requests-oauthlib>=0.7.0 in d:\anaconda\envs\detectron\lib\site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard->detectron2==0.6) (1.3.1)
Requirement already satisfied: importlib-metadata>=4.4 in d:\anaconda\envs\detectron\lib\site-packages (from markdown>=2.6.8->tensorboard->detectron2==0.6) (5.2.0)
Requirement already satisfied: certifi>=2017.4.17 in d:\anaconda\envs\detectron\lib\site-packages (from requests<3,>=2.21.0->tensorboard->detectron2==0.6) (2022.12.7)
Requirement already satisfied: idna<4,>=2.5 in d:\anaconda\envs\detectron\lib\site-packages (from requests<3,>=2.21.0->tensorboard->detectron2==0.6) (3.4)
Requirement already satisfied: charset-normalizer<3,>=2 in d:\anaconda\envs\detectron\lib\site-packages (from requests<3,>=2.21.0->tensorboard->detectron2==0.6) (2.1.1)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in d:\anaconda\envs\detectron\lib\site-packages (from requests<3,>=2.21.0->tensorboard->detectron2==0.6) (1.26.13)
Requirement already satisfied: MarkupSafe>=2.1.1 in d:\anaconda\envs\detectron\lib\site-packages (from werkzeug>=1.0.1->tensorboard->detectron2==0.6) (2.1.1)
Requirement already satisfied: filelock in d:\anaconda\envs\detectron\lib\site-packages (from huggingface-hub->timm->detectron2==0.6) (3.9.0)
Requirement already satisfied: zipp>=3.1.0 in d:\anaconda\envs\detectron\lib\site-packages (from importlib-resources->hydra-core>=1.1->detectron2==0.6) (3.11.0)
Requirement already satisfied: pywin32>=226 in d:\anaconda\envs\detectron\lib\site-packages (from portalocker->iopath<0.1.10,>=0.1.7->detectron2==0.6) (305.1)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in d:\anaconda\envs\detectron\lib\site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard->detectron2==0.6) (0.4.8)
Requirement already satisfied: oauthlib>=3.0.0 in d:\anaconda\envs\detectron\lib\site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard->detectron2==0.6) (3.2.2)
Installing collected packages: detectron2
  Running setup.py develop for detectron2
    error: subprocess-exited-with-error

    × python setup.py develop did not run successfully.
    │ exit code: 1
    ╰─> [21 lines of output]
        running develop
        running egg_info
        writing detectron2.egg-info\PKG-INFO
        writing dependency_links to detectron2.egg-info\dependency_links.txt
        writing requirements to detectron2.egg-info\requires.txt
        writing top-level names to detectron2.egg-info\top_level.txt
        reading manifest file 'detectron2.egg-info\SOURCES.txt'
        adding license file 'LICENSE'
        writing manifest file 'detectron2.egg-info\SOURCES.txt'
        running build_ext
        building 'detectron2._C' extension
        creating build\temp.win-amd64-cpython-38\Release\University\FYP\OUR FYP\Model Training\Non Augmented Dataset MODELS\Model3 Detectron2 412\detectron2\detectron2
        D:\Anaconda\envs\detectron\lib\site-packages\setuptools\command\easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
          warnings.warn(
        D:\Anaconda\envs\detectron\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
          warnings.warn(
        D:\Anaconda\envs\detectron\lib\site-packages\torch\utils\cpp_extension.py:369: UserWarning: Attempted to use ninja as the BuildExtension backend but we could not find ninja.. Falling back to using the slow distutils backend.
          warnings.warn(msg.format('we could not find ninja.'))
        D:\Anaconda\envs\detectron\lib\site-packages\torch\utils\cpp_extension.py:304: UserWarning: Error checking compiler version for cl: [WinError 2] The system cannot find the file specified
          warnings.warn(f'Error checking compiler version for {compiler}: {error}')
        error: could not create 'build\temp.win-amd64-cpython-38\Release\University\FYP\OUR FYP\Model Training\Non Augmented Dataset MODELS\Model3 Detectron2 412\detectron2\detectron2': The filename or extension is too long
        [end of output]

    note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× python setup.py develop did not run successfully.
│ exit code: 1
╰─> [21 lines of output]
    running develop
    running egg_info
    writing detectron2.egg-info\PKG-INFO
    writing dependency_links to detectron2.egg-info\dependency_links.txt
    writing requirements to detectron2.egg-info\requires.txt
    writing top-level names to detectron2.egg-info\top_level.txt
    reading manifest file 'detectron2.egg-info\SOURCES.txt'
    adding license file 'LICENSE'
    writing manifest file 'detectron2.egg-info\SOURCES.txt'
    running build_ext
    building 'detectron2._C' extension
    creating build\temp.win-amd64-cpython-38\Release\University\FYP\OUR FYP\Model Training\Non Augmented Dataset MODELS\Model3 Detectron2 412\detectron2\detectron2
    D:\Anaconda\envs\detectron\lib\site-packages\setuptools\command\easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    D:\Anaconda\envs\detectron\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    D:\Anaconda\envs\detectron\lib\site-packages\torch\utils\cpp_extension.py:369: UserWarning: Attempted to use ninja as the BuildExtension backend but we could not find ninja.. Falling back to using the slow distutils backend.
      warnings.warn(msg.format('we could not find ninja.'))
    D:\Anaconda\envs\detectron\lib\site-packages\torch\utils\cpp_extension.py:304: UserWarning: Error checking compiler version for cl: [WinError 2] The system cannot find the file specified
      warnings.warn(f'Error checking compiler version for {compiler}: {error}')
    error: could not create 'build\temp.win-amd64-cpython-38\Release\University\FYP\OUR FYP\Model Training\Non Augmented Dataset MODELS\Model3 Detectron2 412\detectron2\detectron2': The filename or extension is too long
    [end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.

-------------------------------------------
'1.8.0+cu101' is the torch version of my environment

**Comment by nitinmukesh** [comment]
I am facing the same problem.
